design/baremetal-operator/noop-management.md

```diff
@@ -0,0 +1,160 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Add support for Ironic noop management driver in metal3
+
+## Status
+
+implementable
+
+## Summary
+
+This design aims to provide the operator with a metal3 interface
+to disable Ironic BIOS settings management for a specific baremetal node.
+This feature is already available in Ironic for `IPMI` and `Redfish` drivers
+and is implemented as `noop` `management_interface` for those drivers. Metal3
+currently doesn't support this feature.
+
+## Motivation
+
+Ironic management interface features include (but are not limited to) making
+changes to the boot order of a node which is being provisioned. In some
+circumstances it is desired to disable this functionality and ensure that the
+BIOS settings (and specifically the boot sequence) on the node do not change.
+
+On certain hardware types, requesting PXE boot via the BMC gives different
+results than expected. For example, PXE boot request may always result in boot
+attempt on the first port of the on-board NIC only, independently of the boot
+order defined by the operator who may wish to boot off a port on an add-on PCI
+network card. If the hardware behaves in such a way, it may be better to have
+a static boot order configuration, hence it is important to provide the
+operators with such capability.
+
+Another use case for this feature is a scenario where the operator wants to
+deploy a certain BIOS configuration and ensure that it remains unchanged.
+This may be imposed by a security policy, a need for a workaround to
+a specific problem or perhaps another reason.
+
+### Goals
+
+Add an extra attribute to `IPMI` and `Redfish` BMC definitions allowing
+creation of Ironic nodes with noop management interface through metal3.
+
+### Non-Goals
+
+- Managing the boot sequence of the node.
+- Ensuring that the node correctly follows operator's configuration.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a user, I want to be able to statically configure boot sequence of a node:
+
+- boot from local storage device first
+- if this fails, boot from the desired network interface
+
+So that I can control exactly which NIC and which port will be used for
+network boot independently on how BMC network boot request works on
+a particular machine.
+
+#### Story 2
+
+As a user, I want to have full control of BIOS settings in use, so that I can
+ensure compliance with security requirements and/or ensure I use the exact
+settings that suit my needs.
+
+## Design Details
+
+Add a new `NoopManagement` attribute to the definitions of BMCs which support
+noop management interface. Similar changes will be required in the Custom
+Resource Definition, `baremetalhost_types`, `make-bm-worker` and BMO
+templates so that the request to create the node with `noop`
+`management_interface` is passed to Ironic.
+
```

Comment:
This section covers the implementation approach, but it'd be useful to add an example to clearly illustrate the new BMH interface.

Reply:
Thank you for your feedback Steve and apologies for the delay in responding. Is this example sufficient, or would you like me to provide further information?